### PR TITLE
Add CircleCI test summaries

### PR DESCRIPTION
**Pull request by Sija**
Depends on #8599

TODO:
- [x] successfully generate a test summary (https://circleci.com/gh/crystal-lang/crystal/35618)
- [x] aggregate reports from multiple spec runs (https://circleci.com/gh/crystal-lang/crystal/35680)

image

### CODE REVIEW

**Comment by Sija** [comment]
hmm, CircleCI seems to fail parsing junit xml file:

```
The following errors were encountered parsing test results:

spec/spec.xml
    ParseError at [row,col]:[5043,112] Message: An invalid XML character (Unicode: 0x0) was found in the value of attribute "name" and element is "testcase".
```

Running the test on my local machine and checking line `5043` (`sed -n 5043p output.xml`) yields:

```xml
  <testcase file="/Users/sija/Code/crystal/spec/std/string_spec.cr" classname="Users.sija.Code.crystal.spec.std.string_spec" name="String delete assert"/>
```

**Comment by Sija** [comment]
Rebased on #8599 (https://circleci.com/gh/crystal-lang/crystal/35618)

image

File: bin/ci [hunk 1]
```diff
@@ -73,27 +73,27 @@ prepare_system() {
...
         i=$((i + 1))
       done
 
       with_build_env 'make docs threads=1'
       ;;
     *)
-      with_build_env 'make crystal std_spec compiler_spec docs threads=1'
+      with_build_env 'make crystal std_spec compiler_spec docs threads=1 junit_output=.'
       ;;
   esac
```
Comment (Sija) [inline bin/ci]: This needs to be changed in order to harvest junit output files from all of the runs.
Sth TBD.

Comment (Sija) [reply bin/ci]: Done (https://circleci.com/gh/crystal-lang/crystal/35680)

image

**Comment by Sija** [comment]
GTG

**Comment by straight-shoota** [comment]
Why not put the output files directly in `~/test-results/spec`?

**Comment by Sija** [comment]
```
.build/std_spec: relocation error: .build/std_spec: symbol __gmon_start__ version GLIBC_2.3.3 not defined in file libpthread.so.0 with link time reference
```

🤔

**Comment by asterite** [comment]
It's the latest known failure in linux CI. We should ignore it for now until we figure out what's wrong (maybe it's the number of spec/object files, not sure).

**Comment by Sija** [comment]
> Why not put the output files directly in `~/test-results/spec`?

@straight-shoota I did it because the path is referenced throughout multiple files, some of them docker-related so I wanted to have output scope to the `cwd` and then explicitly copied wherever appropriate within the CI env. Perhaps it can be simplified, feel free to commit into this PR.

**Pull request merged**